### ISSUE TITLE
Why can't I bundle tangram.js in a single dependencies file

### ISSUE THREAD

**Issue by thisisaaronland** [whosonfirst/whosonfirst-www-iamhere#8]
https://github.com/whosonfirst/whosonfirst-www-iamhere/blob/master/Makefile#L3-L4

https://github.com/whosonfirst/whosonfirst-www-iamhere/blob/master/www/index.html#L28-L29
## 

No loading errors but no map rendering either...

**Comment by bcamper** [comment]
This is a known limitation currently (should be documented and/or generate an error it sounds like!). 

This stems from Tangram's worker architecture. Once Tangram's main thread starts, it has to instantiate its worker(s) with either a URL or JS blob. The current build process creates a single bundle, which is loaded into _both_ the main thread and the worker threads. This simplifies the dependency/build process (and the threads share an increasing amount of code anyway), as well as the distribution/hosting -- we could build a second worker-specific bundle (we used to), but then someone wanting to host Tangram remotely would have to host two files (maybe not a big deal but more confusing than one).

Anyway, basically Tangram needs to load itself into a worker. It currently does this by looking for the `<script>` tag from which it was loaded, which it then feeds to a worker. That means that file (even if it was findable) also has to be setup to execute the Tangram worker code when instantiated as a worker. Likely neither is true when it's been wrapped into another bundle.

I do think this should be supported (we found that it also prevents the use of Webpack, even when loading Tangram from a script tag, because Webpack re-writes the document entirely before Tangram can find the script tag it needs!). There is also an option (looks like it would need to be exposed to the Leaflet plugin) to specify a worker URL to load directly, but that doesn't solve your problem here (might as well just host it separately in the first place then).

Perhaps there's a way for Tangram to access its own source without having to reload it from a URL? Like if the actual Tangram bundle loaded Tangram source into a variable, which was then eval'ed. Sounds dicey. Or if it is bundled, perhaps a way for it to extract just itself from the bundle? I don't know if this is possible.
